### DISCUSSION TITLE
I want reopen all Editors(code editor, custom editor, custom text editor mixed open)

### DISCUSSION THREAD

**Discussion by lusian01** [microsoft/vscode-discussions#120] [Extension Development QnA] [answered]
I am a developer, developing extensions of visual studio code.

I was posted on [stackoverflow](https://stackoverflow.com/questions/73554567/reopen-all-editorscode-editor-custom-editor-custom-text-editor-mixed-open)(vscode question),  but I'm sorry for posting it here because it's urgent.
This place was recommended by [vscode github](https://github.com/microsoft/vscode/issues/159725).
So let me ask you a question.

What I want to do is as follows, but I am asking because it seems that the VSCode extension api does not provide the corresponding function.

[Now What i want....]
I opened custom editor, code editor, and custom text editor is opening on vscode.(mixed open)
ex) vue file 'A' using code editor(vscode built-in), vue file 'A' using custom editor,vue file 'B' using code editor(vscode built-in), custom text editor(json file)

I want to close all currently open views by using vscode-extension api and open them again in the order in which they were originally opened.

I checked the vscode API for this, but there doesn't seem to be an api to get all open editor information and an api to reopen all editors. Are there any APIs available?

It was confirmed that "vscode.action.activeEditor" command only gets code editor information.

"view: quick open previous recenlty used editor" commands also confirmed, but it opened commandline, so this is not the result I want.

"window.tabGroups" shows undefined data, so this is not the result what i want.

If there is no corresponding API, can I know if there is a bypass method or plan to add it?

**Comment by gjsjohnmurray** [comment, marked as answer]
Have you played with the sample at https://github.com/microsoft/vscode-extension-samples/tree/main/tabs-api-sample yet?

**Reply by lusian01** [reply]
I checked "vscode.window.tabGroups" marked in tabs-api-sample and it was "undefined".
I can get "vscode.window", but "vscode.window.tabGroups" is undefined.

**Reply by gjsjohnmurray** [reply]
Please start by cloning the extension samples repo, build tabs-api-sample and verify it's working for you. When debugging through it you should be able to see what values are available.

**Reply by lusian01** [reply]
I did this sample, and i tried this sample code in my project(this project is security project, so i can't show the code, sorry.), but it does not work. My guess is that the configuration of the project you're currently developing doesn't support the sample's api.

You gave a good sample but unfortunately it doesn't work for my project.

**Reply by burkeholland** [reply]
Those are the official samples and they _should_ work for all VS Code extensions. They are also examples of the canonical way to use the API. If they don't work for your project, you may have an outdated version of some of the dependencies.
